sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -1002,6 +1006,183 @@ mod test_topology_manager_policy {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// imageGCHighThresholdPercent is the percent of disk usage after which image
+/// garbage collection is always run. The percent is calculated by dividing this
+/// field value by 100, so this field must be between 0 and 100, inclusive. When
+/// specified, the value must be greater than imageGCLowThresholdPercent.
+/// Default: 85
+/// https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ImageGCHighThresholdPercent {
+    // All settings are brought in as Strings.
+    // Important note: custom types need to have `string_impls_for!()` after
+    // their TryFrom<&str> implementation. (this requires that the inner type
+    // of a custom Kubernetes settings type is a String)
+    inner: String,
+}
+
+impl TryFrom<&str> for ImageGCHighThresholdPercent {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        // apiserver validation 1: reject empty values
+        ensure!(
+            !input.is_empty(),
+            error::InvalidImageGCHighThresholdPercentSnafu {
+                input,
+                msg: "must not be empty",
+            }
+        );
+        // apiserver validation 2: reject values that are too low to be a percentage
+        ensure!(
+            i32::ge(&input.parse::<i32>().unwrap(), &IMAGE_GC_THRESHOLD_MIN),
```

Comment:
Would this be clearer as `input.parse::<i32>().unwrap() > IMAGE_GC_THRESHOLD_MIN`?

Reply:
Yes, but:

You should avoid using `unwrap()` or `expect()` in almost all cases except for tests, instead transforming an error in parsing into an error with the parent function. Ultimately this would look like.

Basically you should never panic, instead deciding if you want to handle an error in some way, or bubble it up as a Result to the caller.

```rust
let parsed_input: i32 = input.parse()
    .context(error::InvalidImageGCHighThresholdPercentSnafu { input, msg: "Could not parse input as integer." })?;

ensure!(
    parsed_input > Image_GC_THRESHOLD_MIN,
    // etc
    )
```

Reply:
Update: followed this suggestion (thank you!) and added a new error type for `std::num::ParseIntError`s